youtube_dl/extractor/douyin.py

```diff
@@ -0,0 +1,94 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    float_or_none,
+    try_get,
+    orderedSet
+)
+
+
+class DouyinVideoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?douyin\.com/video/(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'https://www.douyin.com/video/6989098563519270181',
+        'md5': 'b4c99f3be1be343948a9eaaea8303d27',
+        'info_dict': {
+            'id': '6989098563519270181',
+            'url': r're:^https://v3-dy-o.zjcdn.com/[0-9a-f]{32}/[0-9a-f]{8}/video/tos/cn/tos-cn-ve-15/[0-9a-f]{32}/?.*',
+            'title': '杨集#我的家乡 ',
+            'uploader': '永恒的爱',
+            'uploader_id': '104081949894',
+            'timestamp': 1627276320000.0,
+            'ext': 'mp4'
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        iteminfo = self._download_json('https://www.douyin.com/web/api/v2/aweme/iteminfo',
+                                       video_id, query={'item_ids': video_id})
+        if iteminfo['status_code']:
```

Comment:
At this point `iteminfo` isn't known to be a dict. If it isn't, or it has no key `status_code`, this will crash, when it might be preferable to catch all these potential failures in the ExtractorError.
```
...
        iteminfo = self._download_json('https://www.douyin.com/web/api/v2/aweme/iteminfo',
                                       video_id, query={'item_ids': video_id}) or {}
        status_code = iteminfo.get('status_code', 'status_code missing')
        if status_code:
            raise ExtractorError(iteminfo.get('status_msg', 'status_msg missing (%s)' % (status_code, ), video_id=video_id)
...
```